src/api/compare.cc

```diff
@@ -235,5 +248,129 @@ common::ComparisonResult<LaneEnd> IsLaneEndEqual(const LaneEnd& lane_end1, const
   return {std::nullopt};
 }
 
+common::ComparisonResult<bool> IsEqual(const char* a_expression, const char* b_expression, bool a, bool b) {
+  if (a != b) {
+    return {"Values are different. " + std::string(a_expression) + ": " + std::to_string(a) + " vs. " +
+            std::string(b_expression) + ": " + std::to_string(b) + "\n"};
+  }
+  return {std::nullopt};
+}
+
+common::ComparisonResult<double> IsEqual(const char* a_expression, const char* b_expression, double a, double b) {
+  const double delta = std::abs(a - b);
+  if (delta > 1e-6) {
```

Comment:
In some places we are using 1e-12 or below. Do we need this threshold?
Actually, do we need these overloads with basic types?

Reply:
I will change it to at least a `numeric_limits::epsilon`. I don't want to change behavior in downstream tests.